integration-tests/opentelemetry-vertx/src/main/java/io/quarkus/it/opentelemetry/vertx/SpanData.java

```diff
@@ -1,28 +1,7 @@
 package io.quarkus.it.opentelemetry.vertx;
 
-import io.quarkus.runtime.annotations.RegisterForReflection;
+import io.quarkus.runtime.annotations.RegisterForSerialization;
 
-@RegisterForReflection(classNames = {
-        "io.opentelemetry.sdk.trace.data.SpanData",
-        "io.opentelemetry.sdk.trace.SpanWrapper",
-        "io.opentelemetry.sdk.trace.AutoValue_SpanWrapper",
-        "io.opentelemetry.sdk.trace.data.StatusData",
-        "io.opentelemetry.sdk.trace.data.ImmutableStatusData",
-        "io.opentelemetry.sdk.trace.data.AutoValue_ImmutableStatusData",
-        "io.opentelemetry.api.trace.SpanContext",
-        "io.opentelemetry.api.internal.ImmutableSpanContext",
-        "io.opentelemetry.api.internal.AutoValue_ImmutableSpanContext",
-        "io.opentelemetry.api.trace.TraceFlags",
-        "io.opentelemetry.api.trace.ImmutableTraceFlags",
-        "io.opentelemetry.api.trace.TraceState",
-        "io.opentelemetry.api.trace.ArrayBasedTraceState",
-        "io.opentelemetry.api.trace.AutoValue_ArrayBasedTraceState",
-        "io.opentelemetry.sdk.common.InstrumentationLibraryInfo",
-        "io.opentelemetry.sdk.common.AutoValue_InstrumentationLibraryInfo",
-        "io.opentelemetry.sdk.resources.Resource",
-        "io.opentelemetry.sdk.resources.AutoValue_Resource",
-        "io.opentelemetry.api.common.Attributes",
-        "io.quarkus.opentelemetry.runtime.tracing.DelayedAttributes"
-})
+@RegisterForSerialization(classNames = { "io.opentelemetry.sdk.trace.data.SpanData" })
```

Comment:
Good points @zakkak!

Since this integration test aim to test the successful serialization of SpanData class, wouldn't be better to keep the use of the new annotation only? Or am I missing the test goal? (I'm writing this because I haven't gone through the IC results yet)

By the way, with the new annotation, I'm understanding that there is no need for the includeHierarchy attribute into the RegisterForReflection annotation, that's because the original issue was asking for a way to register the full hierarchy for serialization purpose only. 

Actually it was even being implemented by adding this new attribute into RegisterForReflection , but @Sanne helped to understand, [with these comments](https://github.com/quarkusio/quarkus/pull/24474#discussion_r847543455), that a new annotation would be a better way to go. That way, it seems there is no need for the new attribute for RegisterForReflection.

But, what about the serialization attribute from RegisterForReflection annotation? What is it used for today?

Reply:
So, can we agree to change everything back the original naming (RegisterForReflection), so I'll merge back everything to the existent annotation and buildstep, and treat this change as a behavior fix the way you clarified [here](https://github.com/quarkusio/quarkus/pull/24474#discussion_r901625012) @radcortez.

I think is better to let the creation of a new annotation on the feature request that @zakkak pointed out [here](https://github.com/quarkusio/quarkus/pull/24474#pullrequestreview-964522528).

Can I do that way and submit the new changes for CI and review?

Reply:
I have not read all the comments in this thread, but when I was looking at the original issue, my idea was to simply add a flag to `@RegisterForReflection` which would control whether or not the entire hierarchy would be registered (the flag would be `false` by default to preserve the existing behavior).

Reply:
Great! I'm going to perform the changes and submit it until the end of this week.

Reply:
@radcortez the changes are done as specified